core/providers/ruby/ruby.go

```diff
@@ -245,8 +247,28 @@ func (p *RubyProvider) InstallMisePackages(ctx *generate.GenerateContext, miseSt
 		miseStep.Version(ruby, utils.ExtractSemverVersion(string(versionFile)), ".ruby-version")
 	}
 
-	if gemfileVersion := parseVersionFromGemfile(ctx); gemfileVersion != "" {
-		miseStep.Version(ruby, gemfileVersion, "Gemfile")
+	if constraint := extractRubyConstraintFromGemfile(ctx); constraint != "" {
+		allVersions, err := miseStep.Resolver.GetMise().GetAllVersions("ruby", "")
```

Comment:
This is not exactly how the resolver was intended to be used, but I think it is okay?

It might actually be cleaner to use [`b.Resolver.SetVersionAvailable`](https://github.com/railwayapp/railpack/blob/e28b250f38f3071dfcf431b2ca665a5ae6729b31/core/resolver/resolver.go#L88) instead of fetching and comparing the latest version here. It would be something like this

```go
miseStep.SetVersionAvailable(ruby, func(version string) bool {
    return !(strings.Contains(version, "-dev") || strings.Contains(version, "-preview")
})
```

This will better hook into the existing way Railpack checks for available versions